src/toniarts/openkeeper/world/room/GenericRoom.java

```diff
@@ -618,4 +602,43 @@ public GameObject getPillarObject() {
 
         return null;
     }
+
+    @Nullable
+    private RoomCount getRoomCount() {
+        Map<Integer, String> temp = new HashMap<>();
+        temp.put(3, "%44"); // Portal
+        temp.put(2, "%45"); // Lairs
+        temp.put(4, "%46"); // Hatchery
+        temp.put(1, "%47"); // Treasure
+        temp.put(6, "%48"); // Library
+        temp.put(7, "%49"); // Training Room
+        temp.put(10, "%50"); // Workshop
+        temp.put(9, "%51"); // Guard Rooms
+        temp.put(16, "%53"); // Combat Pit
+        temp.put(12, "%54"); // Torture
+        temp.put(11, "%58"); // Prison
+        temp.put(14, "%61"); // Graveyard
+        temp.put(13, "%62"); // Temple
+        temp.put(15, "%63"); // Casino

```

Comment:
Not very good, but if you need to do this, at least make the hash static instead of always creating a new object.